packages/analyzer/src/utils/imports.js

```diff
@@ -17,4 +17,9 @@ export function hasNamedImport(node) {
 /** @example import * as name from './my-module.js'; */
 export function hasAggregatingImport(node) {
   return !!node?.importClause?.namedBindings?.name && !hasNamedImport(node);
-}
\ No newline at end of file
+}
+
+/** @example import './my-module.js'; */
+export function hasSideEffectImport(node) {
+  return 'importClause' in node && node.importClause == null;
```

Comment:
took some jiggling to get this just right. i wonder whether an early exit in the plugin like this would make things a little simpler? that way you at least can be guaranteed you're dealing with an import from there on

```js
if!(ts.isImportDeclaration(node)) {
  return
}
```

Reply:
Ah sorry, only mentioned your comment after I merged. Yeah I think we can improve on this in a separate PR I guess